### PR TITLE
[hue] Added support for different color temperature capabilities and added Channel to set value in Kelvin

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added support for different color temperature capabilities
- Added Channel to set value in Kelvin

A lot of bulbs support different color temperature value ranges. The Hue API exposes those values and thus they can be used by the binding to calculate the ranges individually. Additionally a new Channel for controlling the color temperature in Kelvin has been added.

Depends on https://github.com/openhab/openhab-core/pull/2157

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml [hunk 1]
```diff
@@ -11,6 +11,14 @@
 		<state readOnly="true" pattern="%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS"/>
 	</channel-type>
 
+	<channel-type id="ct" advanced="true">
+		<item-type>Number</item-type>
+		<label>Color Temperature</label>
+		<description>This channel supports adjusting the color temperature in Kelvin.</description>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:17]: As we are adding this channel type to more and more bindings: Wouldn't it make sense to also define it as a system channel, just next to the "normal" color temperature channel type?

Comment (cweitkamp) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:17]: Yes, absolutely. This way of controlling the color temperature has some advantages over the "normal" color temperature channel. I don't know if you followed the discussion in https://github.com/openhab/openhab-addons/issues/8260#issuecomment-761846845, the question is if we really want to continue supporting the Dimmer-based color temperature via system default channel type.

Do you think a UoM based item-type (`Number:Temperature` or new one `Number:ColorTemperature`) is conceivable? Or does it lead to misunderstandings - in general only Kelvin is valid.

Comment (kaikreuzer) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:17]: No, I hadn't followed that discussion, I've now left a comment on it.

> Do you think a UoM based item-type (Number:Temperature or new one Number:ColorTemperature) is conceivable? Or does it lead to misunderstandings - in general only Kelvin is valid.

`Number:Temperature` is imho pretty misleading. `ColorTemperature` is no existing dimension (afaik), so as the value always only makes sense in Kelvin, I guess a plain Number channel should be sufficient.

Comment (cweitkamp) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:17]: I submitted https://github.com/openhab/openhab-core/pull/2157 to add it.

**Review by cpmeister — APPROVED** [review]
LGTM

**Pull request merged**